server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CachedBlobStore.java

```diff
@@ -264,10 +266,12 @@ private boolean isAbleToCache(byte[] bytes) {
     }
 
     private Mono<InputStream> readFromBackend(BucketName bucketName, BlobId blobId) {
-        return Mono.fromCallable(() -> backend.read(bucketName, blobId));
+        return Mono.from(metricFactory.runPublishingTimerMetric(BLOBSTORE_BACKEND_LATENCY_METRIC_NAME,
+            Mono.fromCallable(() -> backend.read(bucketName, blobId))));
     }
 
     private Mono<byte[]> readBytesFromBackend(BucketName bucketName, BlobId blobId) {
-        return Mono.from(backend.readBytes(bucketName, blobId));
+        return Mono.from(metricFactory.runPublishingTimerMetric(BLOBSTORE_BACKEND_LATENCY_METRIC_NAME,
+            () -> backend.readBytes(bucketName, blobId)));
```

Comment:
you are running the metric on a supplier of mono, not the mono itself.

Please write:

```  
  private Mono<byte[]> readBytesFromBackend(BucketName bucketName, BlobId blobId) {
        return Mono.from(metricFactory.runPublishingTimerMetric(BLOBSTORE_BACKEND_LATENCY_METRIC_NAME,
            backend.readBytes(bucketName, blobId)));

```

Reply:
This metricFactory method with either Supplier or Publisher is bad as it makes it easy to write buggy code.

Thanks for falling in the trap.

I propose we change this for it not to happen in the future.

Maybe we should rename `runPublishingTimerMetric` into `decoratePublisherWithTimerMetric` & `decorateSupplierWithTimerMetric` so that we can't do this mistake anymore.

Other ideas / suggestions?

Reply:
> you are running the metric on a supplier of mono, not the mono itself.

Yes, the first tried it fail, i will take another, if fail again i would need some help.